### PR TITLE
Add polygon label placement options

### PR DESCRIPTION
**Pull request by beroso**
Saw [this PR](https://github.com/fleaflet/flutter_map/pull/1220) about centered labels for polygons today.

I think it can be useful to add an option to place the label using [Dart polylabel](https://github.com/beroso/dart_polylabel). It works well on both convex and concave polygons.

Example:

| Centroid (current approach)  | Polylabel |
| ------------- | ------------- |
| Screen Shot 2022-05-05 at 17 44 31 | Screen Shot 2022-05-05 at 17 44 07 |

### CODE REVIEW

**Comment by JaffaKetchup** [comment]
Thanks for your PR! I'll have a proper look tomorrow :)
Just for now though, I can see that the enumerable names might need changing though, as they are a bit confusing.

File: lib/src/layer/label.dart [hunk 1]
```diff
@@ -1,26 +1,30 @@
...
-          maxDx = max(maxDx, element);
-          minDx = min(minDx, element);
-          return previousValue + element;
-        }) /
-        points.length;
-    var dy = points.map((e) => e.dy).toList().fold<double>(
-            0.0, (previousValue, element) => previousValue + element) /
-        points.length;
+    TextStyle? labelStyle, {
+    PolygonLabelPlacement labelPlacement = PolygonLabelPlacement.centroid,
```
Comment (JaffaKetchup) [inline lib/src/layer/label.dart:14]: I think we should change the default to this new method, but that's just my opinion.

File: lib/src/layer/polygon_layer.dart [hunk 2]
```diff
@@ -26,6 +26,11 @@ class PolygonLayerOptions extends LayerOptions {
   }
 }
 
+enum PolygonLabelPlacement {
+  centroid,
+  polylabel,
+}
```
Comment (JaffaKetchup) [inline lib/src/layer/polygon_layer.dart:32]: I also think we might need to rename these options to make them clearer to other people. Perhaps:

```suggestion
enum PolygonLabelPlacement {
  squareCenter,
  polyCenter,
}
```

**Review by JaffaKetchup — CHANGES_REQUESTED** [review]
This is working great! I've commented below on something(s) that might need changing, but I'm open to suggestions :)

**Review by ibrierley — APPROVED** [review]
I'm fine with this either way. It tests fine. It feels like each option in certain cases is preferable when I look at it visually, so no preference on which is default.

polylabel is maybe ambiguous as an parameter, but I guess it does what it says on the tin (uses whatever polylabel calculates, which could change and not be a center...). So no strong preference either way.

File: lib/src/layer/polygon_layer.dart (hunk 2 above)
Comment (JaffaKetchup) [reply lib/src/layer/polygon_layer.dart:32]: Ok, @ibrierley made a good point on this one, so I won't change it...

**Review by JaffaKetchup — APPROVED** [review]
LGTM

**Pull request merged**